server/db/dbManager.ts

```diff
@@ -24,26 +24,42 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(): Promise<models.IPost[]> {
-		return this.model.find({}).exec();
+	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
+		let posts;
+		if (parent) {
+			posts = await this.model.find({ parent }).sort({ upvotes: -1 }).lean().exec();
+			if (depth < parseInt(process.env.MAX_COMMENT_DEPTH)) {
+				for (const post of posts) {
+					post.children = [];
+					if (depth > 0) {
+						parentObject.children.push(post);
+					}
+					await this.getAllPosts(post._id, depth + 1, post);
+				}
+			}
+		} else {
+			posts = this.model.find({ parent: undefined }).lean().exec();
+		}
+		return posts;
 	};
 
-	async addPost(title: string, body: string, username: string) {
+	async addPost(title: string, body: string, username: string, parent?: mongoose.Types.ObjectId) {
 		const post: models.IPost = await this.model.create({
 			title,
 			body,
 			upvotes: 0,
 			author: username,
-			date: new Date()
+			date: new Date(),
+			...parent && { parent }
 		});
 		return post._id;
 	}
 
 	async getNumberOfPosts(): Promise<number> {
-		return this.model.countDocuments().exec();
+		return this.model.find({ parent: undefined }).countDocuments().exec();
 	}
 
-	async getPostsPage(pageSize: number | string, pageNum: number | string, sort: SortOption = SortOption.DEFAULT): Promise<models.IPost[]> {
+	async getPostsPage(pageSize: number | string, pageNum: number | string, { sort = SortOption.DEFAULT, search = '' }: { sort?: SortOption, search?: string }): Promise<models.IPost[]> {
```

Comment:
`{ sort?: SortOption, search?: string }` - should be the exported type to use it for `filterObject` construction.

Reply:
This isn't changed.

Reply:
oh I thought you meant just exporting the SortOption enum type. I will change it so I export the whole object type

Reply:
ok I have added it in the most recent commit